### PR TITLE
Add execution time for failed test modules

### PR DESCRIPTION
**Pull request by foursixnine**
Currently when a test fails, we don't know how long did it take, or how long did the post fail hooks took, unless we look at the timestamps on the logs, displaying total execution fixes the first problem and printing a message in the logs for the post fail hooks, fixes the second.

VR: http://phobos.qa.suse.de/tests/3857623

```
[2021-12-02T18:15:47.191256+01:00] [debug] ||| finished gpg console (runtime: 527 s)
[2021-12-02T18:15:47.191297+01:00] [debug] ||| post fail hooks runtime: 474 s
```

### CODE REVIEW

**Comment by Martchus** [comment]
The VR doesn't look successful.

**Comment by foursixnine** [comment]
> The VR doesn't look successful.

The clone is: http://phobos.qa.suse.de/tests/3857623, updated the referfence

File: basetest.pm [hunk 1]
```diff
@@ -325,15 +328,26 @@ sub run_post_fail {
         # Read them now to not stumble upon them in the next module.
         $self->get_new_serial_output();
     }
+
     $self->fail_if_running();
+    $self->compute_test_execution_time();
+    my $post_fail_hook_execution_time = execution_time($post_fail_hook_start_time);
+    bmwqemu::diag(sprintf("||| post fail hooks runtime: %d s", $post_fail_hook_execution_time));
```
Comment (kalikiana) [inline basetest.pm:335]: Would you mind adding a unit test for that?

File: basetest.pm [hunk 2]
```diff
@@ -325,15 +328,26 @@ sub run_post_fail {
...
     die $msg . "\n";
 }
 
 sub execution_time { time - shift }
 
+sub compute_test_execution_time {
+    my ($self) = @_;
```
Comment (Martchus) [inline basetest.pm:342]: It would be nice to use signatures in new code.

**Review by Martchus — COMMENTED** [review]
The change is already fully covered by tests but maybe it makes sense to add an additional check somewhere within the tests.

File: basetest.pm (hunk 2 above)
Comment (foursixnine) [reply basetest.pm:342]: @Martchus I will try to add signatures on a separate commit, as no methods have them within this package, let's see how it goes.

File: basetest.pm (hunk 1 above)
Comment (foursixnine) [reply basetest.pm:335]: Done, now we look for the specific pfh message.

File: basetest.pm (hunk 2 above)
Comment (foursixnine) [reply basetest.pm]: We should be good to go then 🏎️

**Pull request merged**